addons/doris/script/entry_point.sh

```diff
@@ -0,0 +1,287 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -eo pipefail
+shopt -s nullglob
+
+# Obtain necessary and basic information to complete initialization
+
+# logging functions
+# usage: doris_[note|warn|error] $log_meg
+#    ie: doris_warn "task may be risky!"
+#   out: 2023-01-08T19:08:16+08:00 [Warn] [Entrypoint]: task may be risky!
+doris_log() {
+    local type="$1"
+    shift
+    # accept argument string or stdin
+    local text="$*"
+    if [ "$#" -eq 0 ]; then text="$(cat)"; fi
+    local dt="$(date -Iseconds)"
+    printf '%s [%s] [Entrypoint]: %s\n' "$dt" "$type" "$text"
+}
+doris_note() {
+    doris_log Note "$@"
+}
+doris_warn() {
+    doris_log Warn "$@" >&2
+}
+doris_error() {
+    doris_log ERROR "$@" >&2
+    exit 1
+}
+
+# check to see if this file is being run or sourced from another script
+_is_sourced() {
+    [ "${#FUNCNAME[@]}" -ge 2 ] &&
+        [ "${FUNCNAME[0]}" = '_is_sourced' ] &&
+        [ "${FUNCNAME[1]}" = 'source' ]
+}
+
+docker_setup_env() {
+    declare -g DATABASE_ALREADY_EXISTS
+    if [ -d "${DORIS_HOME}/be/storage/data" ]; then
+        DATABASE_ALREADY_EXISTS='true'
+    fi
+}
+
+# Check the variables required for startup
+docker_required_variables_env() {
+  declare -g RUN_TYPE
+  if [ -n "$BUILD_TYPE" ]; then
+      RUN_TYPE="K8S"
+      if [[ $BUILD_TYPE =~ ^([kK]8[sS])$ ]]; then
+          doris_warn "BUILD_TYPE" $BUILD_TYPE
+      else
+          doris_error "BUILD_TYPE rule error！example: [k8s], Default Value: docker"
+      fi
+      export RUN_TYPE=${RUN_TYPE}
+      return
+  fi
+
+  if [[ -n "$FE_SERVERS" && -n "$BE_ADDR" ]]; then
+      RUN_TYPE="ELECTION"
+      if [[ $FE_SERVERS =~ ^.+:[1-2]{0,1}[0-9]{0,1}[0-9]{1}(\.[1-2]{0,1}[0-9]{0,1}[0-9]{1}){3}:[1-6]{0,1}[0-9]{1,4}(,.+:[1-2]{0,1}[0-9]{0,1}[0-9]{1}(\.[1-2]{0,1}[0-9]{0,1}[0-9]{1}){3}:[1-6]{0,1}[0-9]{1,4})*$ || $FE_SERVERS =~ ^.+:([0-9a-fA-F]{1,4}:){7,7}([0-9a-fA-F]{1,4}|:)|([0-9a-fA-F]{1,4}:){1,6}(:[0-9a-fA-F]{1,4}|:)|([0-9a-fA-F]{1,4}:){1,5}((:[0-9a-fA-F]{1,4}){1,2}|:)|([0-9a-fA-F]{1,4}:){1,4}((:[0-9a-fA-F]{1,4}){1,3}|:)|([0-9a-fA-F]{1,4}:){1,3}((:[0-9a-fA-F]{1,4}){1,4}|:)|([0-9a-fA-F]{1,4}:){1,2}((:[0-9a-fA-F]{1,4}){1,5}|:)|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6}|:)|:((:[0-9a-fA-F]{1,4}){1,7}|:)$ ]]; then
```

Comment:
please don't copy the code and move this check to a separate function

Reply:
sorry，Do you mean that this part of the doris script changes should be written in the mirror instead of exposed changes

Reply:
entry_point.sh is doris official be and cn execute scripts, these codes are not exactly the same, I just reduce some redundant judgment, speed up the startup